### PR TITLE
Add diligent academy sql challenge

### PR DESCRIPTION
**Pull request by dil-ddaradics**

### CODE REVIEW

File: diligent-academy-assessments/backend/database-challenge/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+
+2. Create the class table, each class has a name, and it have to be **unique**.
+   Every student can be in only one class.
+   How would you modify the table if we had a lot of read operations by searching on the class name.
+
+3. How would you query every student in a given class by name?
+
+4. Create the subject table, each subject has a name. Every student can attend multiple subjects.
+   Where would you store the grades of the students?
```
Comment (AkosKovacs0) [inline diligent-academy-assessments/backend/database-challenge/README.md:20]: Should the subject be part of a class or totally freestanding.

File: diligent-academy-assessments/backend/database-challenge/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+5. How would you get the average grade of a students for a given subject by name?
```
Comment (AkosKovacs0) [inline diligent-academy-assessments/backend/database-challenge/README.md:22]:
```suggestion
5. How would you get the average grade of a students for a given subject by the subject's name?
```

**Comment by AkosKovacs0** [comment]
We could also mention referential integrity in some way. And that we require PostgreSQL (do we?).

File: diligent-academy-assessments/backend/database-challenge/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+## Tasks
+
+1. Create the student table with the following fields: name and email. What fields would you create, what would be the
+   type of them?
+
+
+2. Create the class table, each class has a name, and it have to be **unique**.
+   Every student can be in only one class.
+   How would you modify the table if we had a lot of read operations by searching on the class name.
```
[the remark this answers is not in the document]
Comment (dil-ddaradics) [reply diligent-academy-assessments/backend/database-challenge/README.md:15]: Very true, let me rephrase it.

File: diligent-academy-assessments/backend/database-challenge/README.md (hunk 1 above)
Comment (dil-ddaradics) [reply diligent-academy-assessments/backend/database-challenge/README.md:20]: Maybe we are thinking about different things when we talk about classes. I'm thinking about the specific class you are in in a given grade, like how I was once in 9th grade and in class C. I could not find a better word the class.
Subjects are like math and english and stuff.

Comment (AkosKovacs0) [reply diligent-academy-assessments/backend/database-challenge/README.md:20]: I see, that's okay.

File: diligent-academy-assessments/backend/database-challenge/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Database challenge
+
+## Principles and rules
+
+- Please use the prepared `.sql` files to solve the tasks.
+
+## Tasks
+
+1. Create the student table with the following fields: `name` and `email`. What columns would you create to represent
+   them? Which types and constraints would make sense for a user table?
```
Comment (AkosKovacs0) [inline diligent-academy-assessments/backend/database-challenge/README.md:10]: Ooops, I made a mistake here. It should refer to the `students` table not the `user` table:
```suggestion
1. Create the student table with the following fields: `name` and `email`. What columns would you create to represent
   them? Which types and constraints would make sense for this table?
```

**Comment by dil-ddaradics** [comment]
> We could also mention referential integrity in some way. And that we require PostgreSQL (do we?).

I think having foreign keys are enough for them at this level, and I don't think we should use any postgre specific feature. plain is good for me.

File: diligent-academy-assessments/backend/coding-challenge/test/user.spec.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
+import {test, describe, expect} from "@jest/globals";
+import {Book} from "../src/sealed/book";
+import {Library} from "../src/sealed/library";
+import {User} from "../src/implementations/user";
+
+describe('User', () => {
+    describe('can', () => {
```
Comment (dil-kpakozdi) [inline diligent-academy-assessments/backend/coding-challenge/test/user.spec.js:7]: Formatting issues here.

**Review by dil-kpakozdi — DISMISSED** [review]

**Review by dil-ijuhasz — DISMISSED** [review]
